### PR TITLE
Instagram

### PR DESCRIPTION
**Pull request by davidpmccormick**
## What is this PR trying to achieve?
Adding an Instagram post type.

## What does it look like?
screen shot 2017-02-03 at 13 10 35

No JS:
screen shot 2017-02-03 at 13 13 09

## For interface components
- [x] Browser testing - Have you checked the component in our supported browsers
- [x] No javascript - Have you checked the component with javascript disabled
- [x] A11y testing - Have you run pa11y against the component and fixed/verified all errors, warnings and notices? (iFrame title from Instagram empty – outwith our control).

## Have the following been considered/are they needed?
- [x] Tests?

### CODE REVIEW

File: server/util/body-parser.js [hunk 1]
```diff
@@ -178,6 +181,22 @@ export function convertWpList(node) {
   }
 }
 
+function convertInstagramPost(node) {
+  const className = node.attrs && getAttrVal(node.attrs, 'class');
+  const isInstagramPost = Boolean(className && className.match('instagram-media'));
+
+  if (isInstagramPost) {
+    return new BodyPart({
+      type: 'instagramPost',
```
Comment (jamesgorrie) [inline server/util/body-parser.js]: They call it an embed, so shall we call it one?

File: client/js/components/instagram.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+const instagramPost = document.querySelector('.instagram-media');
```
Comment (jamesgorrie) [inline client/js/components/instagram.js]: Why not put this in the function?

**Comment by jamesgorrie** [comment]
This is getting very close to components being fully fledged pieces of content that can import their own styles, DOM and scripts... Let's make sure we think about this if it comes up again.

File: client/js/components/instagram.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
-const instagramPost = document.querySelector('.instagram-media');
-
 function init() {
-  if (instagramPost) {
+  const instagramEmbed = document.querySelector('.instagram-media');
```
Comment (jamesgorrie) [inline client/js/components/instagram.js]: Being really pedantic here - but is it clearer as:
```JS
const hasInstagramEmbed = Boolean(document.querySelector('.instagram-media');
```

I'm going type mad.

**Comment by jamesgorrie** [comment]
You have conflicts

File: client/js/components/instagram.js (hunk 3 above)
Comment (davidpmccormick) [inline client/js/components/instagram.js]: Sorry, missed this – but yeah, I get you. Will fix.

**Pull request merged**